osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -756,7 +755,7 @@ public CarouselRoot(BeatmapCarousel carousel)
 
             protected override void PerformSelection()
             {
-                if (LastSelected == null)
+                if (LastSelected == null || LastSelected.Filtered.Value)
```

Comment:
I would honestly consider carousel selecting the next beatmap when refining your filter a feature and wouldn't change it. I can see a similar scenario to what I laid out in #2638 happening if you refine your current filter while your currently selected map would get filtered out by your new filter.

Reply:
I don't follow.

Reply:
No. The selection happens in SelectNextRandom or in base.PerformSelection. This change (on this line at least) is making carousel select a random beatmap when there hasn't been a selection or when the last selection is filtered.

Reply:
Correct. That's the intention.

Reply:
I did, that's #2638. The change on this line is making the carousel select something away from beatmaps near previous selection and instead select a random one. "Fix carousel not returning to previous selection after filter" is literally the title of the PR, I fail to see how this is out of scope.

Reply:
Please wait until this fix is merged and then rebase your proposed change. I think you are misunderstanding this PR (please see the provided tests).

Reply:
So I think @LittleEndu has an issue with behaviour tested in `RandomFallbackOnNonMatchingPrevious` - the reasoning is that when the previously-selected beatmap is filtered, the carousel should make a "local" change (as in choose from maps in proximity to the filtered ones) instead of making a random one. Whether that's the behaviour we want in the end is disputable and sort of up to preference, so I can't say anything objective about this. Subjectively I don't really care either way myself.

Reply:
Yeah, I think I get that. Not sure it's correct though, and the case I mention in his new PR (where do you draw a line) is probably what we want to consider when deciding a path forward. ie. if 30000 beatmaps are hidden from a filter, do you just select the first remaining one? seems potentially worse than random. why even have random if it's only going to select once ever?

i dunno.